### DISCUSSION TITLE
Upgrade problem — not sure where to start

### DISCUSSION THREAD

**Discussion by jeffkruczynski** [jasonacox/Powerwall-Dashboard#385] [General]
Hello,
I have been running Powerwall-Dashboard on Synology for about a year and have successfully used upgrade.sh in the past, but when trying to run it today, I'm getting this output and error:

```
Upgrade Powerwall-Dashboard from 2.9.10 to 2.10.0
---------------------------------------------------------------------
This script will attempt to upgrade you to the latest version without
removing existing data. A backup is still recommended.

Upgrade - Proceed? [y/N] y

Resetting Timezone to Default...

Pull influxdb.sql, dashboard.json, telegraf.conf, and other changes...

No local changes to save
/opt/lib/git-core/git-remote-https: /opt/lib/libc.so.6: version 'GLIBC_2.33' not found (required by /usr/lib/libcrypto.so.1.1)
/opt/lib/git-core/git-remote-https: /opt/lib/libc.so.6: version 'GLIBC_2.34' not found (required by /usr/lib/libcrypto.so.1.1)
```

I am not sure whether this is an environmental problem or an issue with the upgrade script, and searching for information on the errors hasn't yet surfaced anything super helpful. I don't recall any significant DSM changes since the 2.9.10 Powerwall-Dashboard release that might've affected this, but it's hard to tell. 

Any suggestions on where I should start with this? If it might be an issue with upgrade.sh, I'm happy to move this to the Issues section, but my gut says this is an environmental thing going on.
Thank you!

**Comment by jeffkruczynski** [comment]
Continuing to dig, it looks like my glibc version is 2.36:

```
GNU C Library (GNU libc) stable release version 2.36.
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
Compiled by GNU CC version 12.2.0.
libc ABIs: UNIQUE IFUNC ABSOLUTE
Minimum supported kernel: 3.2.0
For bug reporting instructions, please see:
<https://www.gnu.org/software/libc/bugs.html>
```

**Reply by jasonacox** [reply]
It looks like that is the part of the upgrade script that attempts a `git pull` command.  Can you check that you can still run `git` commands on your Synology?

**Reply by jeffkruczynski** [reply]
I'm able to run `git --version` successfully:
`git version 2.34.3`
And followed ChatGPT's advice below to make sure git seems to be working — it all worked fine.

> Apart from checking the Git version with `git --version`, you can try a few other commands to ensure that Git is properly functioning on your Synology NAS. Here are some additional commands you can use:
> 
> > 1. **Git Help:**
> >    - Run `git help`. This command should display a list of common Git commands and an overview of Git's functionality. If Git is working, you'll see a help message.
> > 
> > 2. **Initialize a Test Git Repository:**
> >    - Create a new directory using `mkdir test-git`.
> >    - Navigate to it with `cd test-git`.
> >    - Initialize a new Git repository with `git init`. This command creates a new `.git` directory in your folder, which is used to track changes. If this command is successful, it indicates that Git is operational.
> > 
> > 3. **Check Git Configuration:**
> >    - Run `git config --list`. This will display the current Git configuration settings. If Git is operational, it will list the configuration details.
> > 
> > 4. **Git Status:**
> >    - In your test Git repository (the `test-git` directory), run `git status`. This command shows the status of the repository. For a new repository, it should indicate that it's clean and there's nothing to commit, which means Git is running correctly.
> > 
> > 5. **Create a Test Commit:**
> >    - Create a new file in the repository, e.g., `echo "Test Git" > testfile.txt`.
> >    - Stage the file for commit with `git add testfile.txt`.
> >    - Commit the file to the repository with `git commit -m "Initial commit"`. This will create a new commit in the repository. If this is successful, it means Git can track and manage changes.
> > 
> > If all these commands run without errors and give expected results, it's a good sign that Git is correctly installed and functioning on your Synology NAS. Remember to delete the test repository (`test-git` directory)

**Reply by jeffkruczynski** [reply]
But — then I tried a `git pull` command and got the same errors. So, it seems more specific to that.
(I did try with a few different repositories, just in case — all had the same error)

**Reply by jasonacox** [reply]
> /opt/lib/git-core/git-remote-https

That seems to indicate it is during the`git pull` part when it must fetch the repo files from the https address of github. This is where the upgrade script failed in your example above, try these command:

```bash
# Pull from GitHub
git stash
git pull --rebase
```

**Reply by jeffkruczynski** [reply]
Ok...on a whim, and because it still felt like something was off with the native git install, I installed the synocommunity git package. The upgrade script just successfully finished!
So, something must've changed with respect to the git version that's installed with DSM. 

Thanks for getting me there!

**Reply by jasonacox** [reply]
That's great news! Good job @jeffkruczynski .

**Reply by billraff** [reply]
@ jeffkruczynski could you point me to the git you installed? I’m also on Synology and when I last tried the upgrade script got errors so gave up. Maybe it will fix my errors as well.

**Reply by jeffkruczynski** [reply]
Hi @billraff — it's the one from SynoCommunity. Do you already have that repository linked in your Package Center? If so: I think there's only one. If you're not finding it, let me know.

**Reply by billraff** [reply]
Yes, found it thanks. Started a new thread here about my new problem.

**Comment by jeffkruczynski** [comment]
For anyone who's on a Synology and stumbles across this, I ran into this issue again today when upgrading from 4.5.5 to 4.6.0.
After some poking around, I found I had two git versions installed on the Synology: one at `/usr/bin/git` and one at `/opt/bin/git` (probably there b/c I installed `opkg`).
I redirected PATH to move the `/usr/bin/git` version (which is 2.39.1, versus 2.34.1) first, and the upgrade script ran fine.

**Reply by jasonacox** [reply]
Thanks @jeffkruczynski ! I added that tip to our README https://github.com/jasonacox/Powerwall-Dashboard?tab=readme-ov-file#troubleshooting-tips-and-tricks
